src/openforms/payments/tests/factories.py

```diff
@@ -8,12 +8,15 @@
 
 
 class SubmissionPaymentFactory(factory.django.DjangoModelFactory):
-    submission = factory.SubFactory(SubmissionFactory)
+    submission = factory.SubFactory(
+        SubmissionFactory, with_public_registration_reference=True
+    )
     plugin_id = "demo"
     plugin_options = {"foo": 123}
-    order_id = factory.Sequence(lambda n: int(f"2020{n:05}"))
     amount = Decimal("10.00")
-    public_order_id = factory.LazyAttribute(lambda obj: f"ref{obj.order_id}")
+    public_order_id = factory.LazyAttributeSequence(
+        lambda obj, n: f"{obj.submission.public_registration_reference}_{n}"
+    )
```

Comment:
can't we invoke a method/function instead that implements the real behaviour? There are things like `PostGenerationMethodCall` available.

Reply:
`PostGenerationMethodCall` only allows method on the model itself. The thing is the real behavior will call `GlobalConfiguration.get_solo()`, so I'm not sure this is a desirable side effect

Reply:
I'm okay with wrapping it in a `unittest.mock.patch` CM. Via LazyAttribute this should be possible :)